src/NumericalAlgorithms/RootFinding/GslMultiRoot.hpp

```diff
@@ -0,0 +1,90 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+// \file
+// Defines function GslMultiRoot.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <exception>
+
+/// cond
+template <size_t Dim>
+class GslMultiRootFunctionFdf;
+/// endcond
+
+/*!
+ *  \ingroup NumericalAlgorithmsGroup
+ *  \brief The exception returned when GslMultiRoot fails to converge.
+ */
+class rootfinding_exception : public std::exception {
+  virtual const char* what() const throw() { return "GslMultiRoot failed."; }
+};
+
+/*!
+ *  \ingroup NumericalAlgorithmsGroup
+ *  \brief The different options for the verbosity of GslMultiRoot.
+ */
+enum class Verbosity {
+  /// Do not print anything
+  Silent,
+  /// Print only "success" or "failed"
+  Quiet,
+  /// Print debug statements.
+  Debug,
+  /// Print everything
+  Verbose
+};
+
+/*!
+ *  \ingroup NumericalAlgorithmsGroup
+ *  \brief The different options for the rootfinding method of GslMultiRoot.
+ */
+enum class Method {
+  /// Hybrid of Newton's method along with
+  /// following the gradient direction.
+  /// Note: Sometimes Hybridsj works only
+  /// with the absolute tolerance termination
+  /// condition.
```

Comment:
The comment in the cpp file is in a section that also requires that the relative tolerance is nonpositive.

Reply:
This has me thinking - instead of implicitly choosing the search stopping criterion through whether the relative_tolerance is positive or not, why dont we have the user explicitly choose through an enum called StoppingCriterion whether they want gsl_multiroot_test_residual or whether they want gsl_multiroot_test_delta called. That way we don't have this indirect "if you want this, you need this other thing..."